File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/domain/interactors/Impl/TryToLoginUseCaseImpl.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.Impl;

import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.TryToLoginUseCase;
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.model.User;
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.repository.Repository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by MateuszW on 2018-05-23.
 */
public class TryToLoginUseCaseImpl implements TryToLoginUseCase {

    private Repository repository;

    @Inject
    public TryToLoginUseCaseImpl(Repository repository) {
        this.repository = repository;
    }

    @Override
    public Observable<User> runUseCase(String... params) {
        return repository.tryLogin(params);
    }
}

File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/domain/interactors/impl/TryToLoginUseCaseImpl.kt
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.impl

import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.TryToLoginUseCase
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.model.User
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.repository.Repository
import io.reactivex.Observable
import javax.inject.Inject

class TryToLoginUseCaseImpl @Inject
constructor(private val repository: Repository) : TryToLoginUseCase {

    override fun runUseCase(vararg params: String): Observable<User> {
        return repository.tryLogin(params[0], params[1])
    }
}
